src/Flysystem/FlysystemAssetStore.php

```diff
@@ -417,21 +418,23 @@ protected function moveBetweenFilesystems($fileID, Filesystem $from, Filesystem
 
     public function grant($filename, $hash)
```

Comment:
Since this discussion we were also thinking of having the current request registered with injector, rather than using Controller::curr().

I've raised https://github.com/silverstripe/silverstripe-framework/issues/7049 as the issue.